.github/workflows/rancher-fleet-ci.yml

```diff
@@ -0,0 +1,224 @@
+name: Rancher Fleet CI
+
+on:
+  schedule:
+    - cron:  '0 5 * * *'
+  workflow_dispatch:
+    inputs:
+      debug_enabled:
+        description: 'Run the build with tmate debugging enabled'
+        required: false
+        default: "false"
+  pull_request:
+    paths-ignore:
+    - 'docs/**'
+    - 'charts/**'
+    - 'scripts/**'
+    - '*.md'
+
+permissions:
+  contents: read
+  packages: write
+
+env:
+  GOARCH: amd64
+  CGO_ENABLED: 0
+  SETUP_GO_VERSION: '^1.18'
+
+jobs:
+  e2e-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      -
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      -
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.SETUP_GO_VERSION }}
+      -
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      -
+        name: Install Ginkgo CLI
+        run: go install github.com/onsi/ginkgo/v2/ginkgo@v2.1
+      -
+        uses: actions/cache@v3
+        id: rancher-cli-cache
+        with:
+          path: /home/runner/.local/bin
+          key: ${{ runner.os }}-rancher-cli-2.6.0
+      -
+        name: Install Rancher CLI
+        if: steps.rancher-cli-cache.outputs.cache-hit != 'true'
+        run: |
+          # download an older CLI to avoid https://github.com/rancher/rancher/issues/37574
+          mkdir -p /home/runner/.local/bin
+          wget -q https://github.com/rancher/cli/releases/download/v2.6.0/rancher-linux-amd64-v2.6.0.tar.gz
+          tar -xz --strip-components=2 -f rancher-linux-amd64-v2.6.0.tar.gz -C /home/runner/.local/bin
+          rancher --version
+      -
+        name: Build fleet binaries
+        run: |
+          go build -o bin/fleetcontroller-linux-$GOARCH ./cmd/fleetcontroller
+
+          go build -o "bin/fleet-linux-$GOARCH"
+          go build -o "bin/fleetagent-linux-$GOARCH" ./cmd/fleetagent
+      -
+        name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      -
+        name: Login to GHCR
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      -
+        name: Docker meta - fleet
+        id: meta-fleet
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            ghcr.io/manno/fleet
```

Comment:
Yes, this workflow needs a registry before it can be green. I asked for github packages in the org, but didn't receive a clear response yet. While Github container registry would be preferred, any registry should do. However, I don't have permissions to set up secrets in the repo, so I can't set up anything that requires them.

As long as we don't make this workflow a blocker for PRs, we can merge it before it actually works, set up the registry and make a small PR to replace the 'ghrc.io/manno' repo.

Reply:
I'm going to suggest this uses GHCR. The tests are running in GitHub Actions. Using the container registry to push/pull from would be best to stay in the same vendor/setup. They can optimize for this.

The images uses here aren't for distribution. So we don't need them to be generally available.